rust/feature-flags/src/flag_matching.rs

```diff
@@ -1102,10 +1130,62 @@ impl FeatureFlagMatcher {
     }
 }
 
-/// Evaluates a single cohort and its dependencies.
+/// Evaluate static cohort filters by checking if the person is in each cohort.
+async fn evaluate_static_cohorts(
+    postgres_reader: PostgresReader,
+    team_id: TeamId,
+    distinct_id: String,
+    cohort_ids: Vec<CohortId>,
+) -> Result<Vec<(CohortId, bool)>, FlagError> {
+    let mut conn = postgres_reader.get_connection().await?;
+
+    let query = r#"
+        WITH person AS (
+            SELECT posthog_person.id AS person_id
+            FROM posthog_person
+            JOIN posthog_persondistinctid 
+              ON posthog_persondistinctid.person_id = posthog_person.id
+            WHERE 
+                posthog_person.team_id = $1
+                AND posthog_persondistinctid.team_id = $1
+                AND posthog_persondistinctid.distinct_id = $2
+            LIMIT 1
+        ),
+        cohort_membership AS (
+            SELECT c.cohort_id, 
+                   CASE WHEN pc.cohort_id IS NOT NULL THEN true ELSE false END AS is_member
+            FROM unnest($3::integer[]) AS c(cohort_id)
+            LEFT JOIN posthog_cohortpeople AS pc
+              ON pc.person_id = (SELECT person_id FROM person)
+              AND pc.cohort_id = c.cohort_id
+        )
+        SELECT cohort_id, is_member
+        FROM cohort_membership
+    "#;
```

Comment:
This query has some extra sauce since I elected to actually parse out the data into the appropriate structures using postgres instead of Rust (it's an easy tuple of `Id`s and `bool`s).  I still need to join on the persondistinctid table, though, since I don't have that relationship at the time I need to evaluate this condition.